### PR TITLE
fix(builder): properly escape backticks

### LINKED ISSUE #2980: config value escaping
I had something like the following config set for DATABASE_URL:

```
$ deis config
=== services Config
DATABASE_URL                postgres://user:EbMCMW0^`ToWS7-v@example.us-east-1.rds.amazonaws.com:5432
```

when I was trying to push I got this error:

```
$ git push deis master
Counting objects: 4, done.
Delta compression using up to 4 threads.
Compressing objects: 100% (3/3), done.
Writing objects: 100% (4/4), 411 bytes | 0 bytes/s, done.
Total 4 (delta 1), reused 0 (delta 0)
/home/git/builder: eval: line 117: unexpected EOF while looking for matching ``'
/home/git/builder: eval: line 118: syntax error: unexpected end of file
To ssh://git@deis.example.net:2222/services.git
 ! [remote rejected] master -> master (pre-receive hook declined)
error: failed to push some refs to 'ssh://git@deis.example.net:2222/services.git'
```

removing the "`" character fixed the issue.

### LINKED ISSUE #3010: Unable to perform deis push
```
Counting objects: 601, done.
Delta compression using up to 8 threads.
Compressing objects: 100% (301/301), done.
Writing objects: 100% (601/601), 199.60 KiB | 0 bytes/s, done.
Total 601 (delta 306), reused 522 (delta 277)
/home/git/builder: eval: line 117: unexpected EOF while looking for matching `"'
/home/git/builder: eval: line 118: syntax error: unexpected end of file
To ssh://git@deis.core.withinboredom.info:2222/old-site.git
 ! [remote rejected] master -> master (pre-receive hook declined)
error: failed to push some refs to 'ssh://git@deis.core.withinboredom.info:2222/old-site.git'
```

deis config:
```
=== old-site Config
AUTH_KEY              26K5,tv*eASx1C%vvyf4Ilg4MSEr?nG0-*r$:MRa$;yAy?VTx1k6#rwl!0D8%9uU
AUTH_SALT             LB9 ?^5VN*Vv/pKP]_1!f~f-UiUM[muuO~dN59.eMPAJ4!{PRMlWCqD[z[vi:PDK
DB_HOST               yogurt-novelist
DB_NAME               wordpress
DB_USER               admin
LOGGED_IN_KEY         xlO%iT[%H |3sq)jl1#Jo@u $ls?3(,<lI2?<L$vt${L~4V-stpG+Ko4WH3llW|b
LOGGED_IN_SALT        tllzFa0<;#G0lBy/uHLO!%-O|tkZlc-Eo!3T@qG7>$taaS/u{J9WjP/i9h8-kOky
MYSQL_PASS            wicked
NONCE_KEY             +1a)?$asM{4CQli8Fxbw8q<qXQC8|B!+~@!>K@g[W@l| u7oS9Km{yCI%4d%=%5_
NONCE_SALT            l%( llQ(]9){n&f|l^WuqTD{1?5#N^/YQ^HG<[l@TlW7726)EoaU8DGvB/,%$+zp
SECURE_AUTH_KEY       NLtJ]!KY69]l7GK+.VX6#)*IFhdg&x0p9)ZUQ:ei5>|zvJqpl$ 3T0?{7b+KzR0.
SECURE_AUTH_SALT      FH[:c(bOuB]{/9APiM?K2Bt>lP/h+ASw7Bt!gukAV.mT4N.#{=jOuCCF(zN90+=u
WP_ENV                staging
WP_HOME               http://old-site.core.withinboredom.info
WP_SITEURL            http://old-site.core.withinboredom.info
```

### PR DESCRIPTION
**Pull request by bacongobbler**
closes #2980
closes #3010

### CODE REVIEW

File: tests/config_test.go [hunk 1]
```diff
@@ -10,7 +10,7 @@ import (
 
 var (
 	configListCmd  = "config:list --app={{.AppName}}"
-	configSetCmd   = "config:set FOO=讲台 --app={{.AppName}}"
+	configSetCmd   = "config:set FOO=讲台` --app={{.AppName}}"
```
Comment (carmstrong) [inline tests/config_test.go]: Could we add a new test for this? I'd rather know that if this test fails, it's an issue with character set encoding and not the backtick escaping.

Comment (bacongobbler) [inline tests/config_test.go]: sure :)

**Comment by mboersma** [comment]
Backticks often show up in tokens / passwords used with `deis config:set`, so this is an important fix. Code LGTM.

**Comment by carmstrong** [comment]
Code LGTM.

**Pull request merged**